### DISCUSSION TITLE
Sortable on accessor column

### DISCUSSION THREAD

**Discussion by gregoryloichot** [filamentphp/filament#13948] [Help]
### Package

Table builder

### Package Version

v3.2

### How can we help you?

Hello, 

I'm trying to make a column sortable, but I get error.

This is my column 👍 
```php
Tables\Columns\TextColumn::make('is_completed')
                    ->label('Status')
                    ->badge()
                    ->sortable('is_completed'),
```
My accessor in the model :
```php
protected function isCompleted(): Attribute
    {
        return Attribute::make(
            get: fn(mixed $value, array $attributes): bool => $this->needed <= $this->posts->count(),
        );
    }
```

The display is completely fine, the values are correct in the column. 

According to the doc (https://filamentphp.com/docs/3.x/tables/columns/getting-started#sorting) this should also work with accessors…

But, when I try to sort the column, I get an SQL error telling me that the column 'is_completed' does not exist (and it is true). So it looks like it ignores the accessor when sorting.

Any ideas ? Did I miss anything ?

**Comment by leandrocfe** [comment]
You can’t use sortable or searchable in accessors because these methods rely on the actual table columns.
You could try to customize this using a callback

```php
->sortable(function (Builder $query) {
    return $query...;
})
```

**Reply by gregoryloichot** [reply]
So, there is no solution ? Any workaround ?
